guides/phoenix/architecture.md

```diff
@@ -42,6 +42,11 @@ defmodule MyApp.OrderService do
   require Ecto.Query
   alias MyApp.{Order, Repo}
 
+# I think this is just a point for discussion. if `Order.released_for_shipment?(order)`
+returns `false`, return value from `with` function also `false`, which could be confusing.
+Maybe add else case which would return `{:error, reason}`? Also, in codebase I think I've seen
+a good pattern being used by returning `{:ok, result}`. I am wondering if we shall reflect that
+good use of with in here?
```

Comment:
Good suggestion here, @tetiana12345678 regarding the result tuple. I've never liked this example because there is a violation of "tell, don't ask" which I feel still applies in functional programming. i.e. we are checking if the order can be released; `Order.released_for_shipment?(order)`. We should call ` Order.acknowledge` and deal with the error if it can't be released, which in this case will be handled by the client calling `acknowledge/2`.

```elixir
def acknowledge(%{"order_id" => order_id}, seller_id) do
  with order = fetch_order(order_id, seller_id)
       {:ok, changeset} <- Order.acknowledge(order),
  do: Repo.update(changeset)
end
```

Reply:
👍 I think this is a good example. I'll add it in favour of current example.